staging/src/k8s.io/client-go/rest/request_test.go

```diff
@@ -488,6 +489,14 @@ func TestURLTemplate(t *testing.T) {
 			ExpectedFullURL:  "http://localhost/some/base/url/path/pre1/namespaces/namespaces/namespaces/namespaces/namespaces/namespaces/finalize",
 			ExpectedFinalURL: "http://localhost/%7Bprefix%7D",
 		},
+		{
+			// dynamic client with core group + namespace + resourceResource (with name) where baseURL is a single /
+			// /api/$RESOURCEVERSION/namespaces/$NAMESPACE/$RESOURCE/%NAME
+			Request: NewRequest(nil, "DELETE", uriSingleSlash, "", ContentConfig{GroupVersion: &schema.GroupVersion{Group: "test"}}, Serializers{}, nil, nil, 0).
+				Prefix("/api/v1/namespaces/ns/r1/name1"),
+			ExpectedFullURL:  "/api/v1/namespaces/ns/r1/name1",
+			ExpectedFinalURL: "api/v1/namespaces/%7Bnamespace%7D/r1/%7Bname%7D",
```

Comment:
actually I'd expect preserving the leading `/`

`ExpectedFinalURL: "/api/v1/namespaces/%7Bnamespace%7D/r1/%7Bname%7D",`

Reply:
@roycaihw I've made some changes for that case. Please take a look again